src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -120,33 +125,68 @@ public SNITCPHandle(string serverName, int port, long timerExpire, object callba
                     ts = ts.Ticks < 0 ? TimeSpan.FromTicks(0) : ts;
                 }
 
-                Task<Socket> connectTask;
-                if (parallel)
-                {
-                    Task<IPAddress[]> serverAddrTask = Dns.GetHostAddressesAsync(serverName);
-                    serverAddrTask.Wait(ts);
-                    IPAddress[] serverAddresses = serverAddrTask.Result;
+                bool reportError = true;
 
-                    if (serverAddresses.Length > MaxParallelIpAddresses)
+                try
+                {
+                    if (parallel)
```

Comment:
We need to add more comments here. parallel I think is true when the MultiSubnetFailover is set to true. While you are there can we add some comments please?

Reply:
I will add the comments for this part. Basically, the connection logic is the same except for giving more tries when the DNS resolution fails for the first time. I wrapped the parts for parallel connection into the `TryConnectParallel` function which will make the retry logic clearer.